server/lib/tuist_web/user_socket.ex

```diff
@@ -0,0 +1,38 @@
+defmodule TuistWeb.UserSocket do
+  use Phoenix.Socket
+
+  alias Tuist.Authentication
+
+  channel("qa_logs:*", TuistWeb.QALogChannel)
+
+  def connect(%{"token" => token}, socket, _connect_info) do
+    case authenticate_socket(token) do
+      {:ok, subject} ->
+        socket = assign(socket, :current_subject, subject)
+        {:ok, socket}
+
+      {:error, _reason} ->
+        :error
```

Comment:
The `connect` callback supports passing the reason, what if we included it here?

```suggestion
      {:error, reason} ->
        {:error, reason}
```